src/python/pants/option/errors.py

```diff
@@ -27,7 +27,7 @@ def __init__(self, scope: str, option: str, **msg_format_args) -> None:
 
 
 class BooleanOptionNameWithNo(RegistrationError):
-    """"Boolean option names cannot start with --no."""
```

Comment:
This one is a bit of a minibugfix. Looks good to me, but may have made sense as a separate commit so it stands out more. Pointing it out so other reviewers can make sure they're fine with removing one quote from the content of the doc comment.